docs/io/ascii/read.rst

```diff
@@ -120,11 +120,14 @@ Parameters for ``read()``
   for more information and examples. The default is that any blank table
   values are treated as missing.
 
-**fill_include_names** : list of column names, which are affected by ``fill_values``.
-  If not supplied, then ``fill_values`` can affect all columns.
+**fill_include_names** : list of column names affected by ``fill_values``.
+  From the list of column names found from the header or the ``names``
+  parameter, select for filling only columns within this list. If not supplied,
+  then apply ``fill_values`` to all columns.
 
-**fill_exclude_names** : list of column names, which are not affected by ``fill_values``.
-  If not supplied, then ``fill_values`` can affect all columns.
+**fill_exclude_names** : list of column names not affected by ``fill_values``.
+  Exclude these names from the list of columns for filling. This is applied *after*
+  the ``fill_include_names`` filtering. If not specified then no columns are excluded.
 
```

Comment:
Is there any reason to even allow `fill_exclude_names` at all if `fill_include_names` is used? `fill_include_names` must be an exhaustive list of all column names to be filtered, so if you want to exclude one of those, just don't list it in `fill_include_names`. Your clarification of the doc string is correct, but it's still complicated. I think form a user perspective it would be much simpler, to use only include or exclude, but not both.

(For backwards compatibility, we might not enforce that really only one of them is used, but if we document it that way, users know that they are on your own if they do so.)

Another approach is to describe more when these options make sense and less how it's acutally done in the code:
```
**fill_include_names** : list of column names affected by ``fill_values``.
  This is a list of column names (found from the header or the ``names``
  parameter) for all columns where values will be filled. `None` (the default) will
  apply ``fill_values`` to all columns.

**fill_exclude_names** : list of column names not affected by ``fill_values``.
  This is a list of column names (found from the header or the ``names``
  parameter) for all columns where values will be **not** be filled. Listing a 
  few columns in `fill_exclude_names` has   the same effect as listing all the 
  other columns in `fill_include_names`. If a column name is listed in both
  `fill_include_names` and `fill_exclude_names`, its values are **not** filled.
  `None` (default) does not exclude any columns.
```
(Note that the formatting is probably screwed up because I type this in the github web form.)

Reply:
This is sort of a common issue, e.g. in rsync you have `--include` and `--exclude` that are mostly useful on their own but are not strictly mutually exclusive.  Generally speaking I would imagine most users would use one or the other, but using both should not be an exception.  An example (perhaps contrived) where you might want to use both is if you have a long list of column names and you want to use fill values for every column starting with `s`, except for the `s1` and `s2 ` column.
```
fill_include_names = [colname for colname in colnames if colname.startswith('s')]
fill_exclude_names = ['s1', 's2']
```
How about your version but with `fill_exclude_names` like:
```
**fill_exclude_names** : list of column names not affected by ``fill_values``.
  This is a list of column names (found from the header or the ``names``
  parameter) for all columns where values will be **not** be filled.
  This parameter takes precedence over ``fill_include_names``.  A value
  of `None` (default) does not exclude any columns.

Reply:
Sounds good.